libraries/AP_NavEKF3/AP_NavEKF3.cpp

```diff
@@ -1049,12 +1057,22 @@ void NavEKF3::checkLaneSwitch(void)
     }
 
     // update the yaw and position reset data to capture changes due to the lane switch
-    if (newPrimaryIndex != primary) {
-        updateLaneSwitchYawResetData(newPrimaryIndex, primary);
-        updateLaneSwitchPosResetData(newPrimaryIndex, primary);
-        updateLaneSwitchPosDownResetData(newPrimaryIndex, primary);
-        primary = newPrimaryIndex;
-        lastLaneSwitch_ms = now;
+    switchLane(newPrimaryIndex);
+}
+
+// switch to a new lane
+void NavEKF3::switchLane(uint8_t new_lane_index)
```

Comment:
is it lane or core?

Reply:
I have kept it as lane to keep it consistent with the nearby code